File: src/foam/nanos/auth/EnabledAware.js
/**
 * @license
 * Copyright 2017 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */

foam.INTERFACE({
  package: 'foam.nanos.auth',
  name: 'EnabledAware',

  methods: [
    {
      name: 'getEnabled',
      type: 'Boolean'
    },
    {
      name: 'setEnabled',
      args: [
        {
          name: 'value',
          type: 'Boolean'
        }
      ]
    }
  ]
});

foam.CLASS({
  package: 'foam.nanos.auth',
  name: 'EnabledAwareDummy',

  properties: [
    {
      class: 'Long',
      name: 'id'
    },
    {
      class: 'Boolean',
      name: 'enabled'
    }
  ]
});

File: src/foam/dao/EnabledAware.java
/**
 * @license
 * Copyright 2017 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */

package foam.dao;

import foam.dao.pg.IndexedPreparedStatement;
import foam.mlang.predicate.Predicate;

import java.util.Date;

public interface EnabledAware {
  public final static Predicate ENABLED = new Predicate() {
    public boolean f(Object obj) { return true; }
    public String createStatement() { return "enabled = 't'"; }
    public void prepareStatement(IndexedPreparedStatement stmt) { }
    public foam.mlang.predicate.Predicate partialEval() { return this; }
  };

  public Date getEnabled();
  public void setEnabled(Date date);
}
